dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/container/config/WebAppConfiguratorHelper.java

```diff
@@ -1447,7 +1467,37 @@ private void configureAdministeredObjects(List<AdministeredObject> adminObjects)
             }
         }
     }
-   
+
+    private void configureContextServices(List<ContextService> contextServices) {
+        Map<String, ConfigItem<ContextService>> configItemMap = configurator.getConfigItemMap("context-service");
+        for (ContextService contextService : contextServices) {
+            String name = contextService.getName();
+            if (name == null) {
+                continue;
+            }
+            ConfigItem<ContextService> existed = configItemMap.get(name);
+            if (existed == null) {
+                configItemMap.put(name, createConfigItem(contextService, CONTEXT_SERVICE_COMPARATOR));
+                webAppConfiguration.addRef(JNDIEnvironmentRefType.ContextService, contextService);
+            } else {
+                if (existed.getSource() == ConfigSource.WEB_XML && configurator.getConfigSource() == ConfigSource.WEB_FRAGMENT) {
+                    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                        Tr.debug(this, tc, "context-service.name with value " + existed.getValue() +
+                                 " is configured in web.xml, the value " + name + " from web-fragment.xml in " +
+                                        configurator.getLibraryURI() + " is ignored");
+                    }
+                } else if (existed.getSource() == ConfigSource.WEB_FRAGMENT && configurator.getConfigSource() == ConfigSource.WEB_FRAGMENT
+                           && !existed.compareValue(contextService)) {
+                    configurator.addErrorMessage(Tr.formatMessage(tc, "SRVE9950_WEB_FRAGMENT_XML_RESOURCE_CONFLICT",
```

Comment:
should it be Tr.getFormattedMessage()  instead of Tr.formatMessage()?

Reply:
When using Tr the method name is formatMessage rather than getFormattedMessage, which doesn't exist there, so this is correct and compiles cleanly.